### PR TITLE
update roles routes

### PR DESCRIPTION
**Pull request by GSansigolo**

### CODE REVIEW

File: bdc_auth/users/views.py [hunk 1]
```diff
@@ -103,4 +103,15 @@ def set_role(user):
...
+
+@bp.route('/role/remove', methods=('POST',))
+@login_required
+def delete_assigned_role(user):
+    """Delete a role-user relation."""
+    try:
+        controller.delete_assigned_role(owner_id=user.id, **request.json)
+    except Exception as e:
+        # TODO treat exceptions and send proper error code.
+        abort(400, '')
```
Comment (raphaelrpl) [inline bdc_auth/users/views.py:116]: Do you really need to handle a individual exception type here?
If so, please use the Exception Type properly and set a message.

We have set a a global catch for server errors in `__init__.py`

File: bdc_auth/clients/controller.py [hunk 2]
```diff
@@ -160,13 +168,16 @@ def get_users_roles_by_client(user_id, client_id):
         .outerjoin(user_roles_mapper, user_roles_mapper.c.id == User.id)\
         .with_entities(*entities)
 
-    if user_id:
-        user = user_query.filter(User.id == user_id).first_or_404()
+    new_user_query = []
 
-        return dict(users=user._asdict())
+    for item in user_query:
```
Comment (gqueiroz) [inline bdc_auth/clients/controller.py:173]: Review

File: bdc_auth/users/controller.py [hunk 3]
```diff
@@ -103,14 +103,24 @@ def update_user(user_id, email=None, name=None, password=None, occupation=None,
...
-    _ = OAuth2Client.query.filter(
-        OAuth2Client.user_id == owner_id,
-        OAuth2Client.id == role.client_id
-    ).first_or_404()
+    with db.session.begin_nested():
+        user.roles.append(role)
+
+    db.session.commit()
 
+def delete_assigned_role(owner_id, user_id, role_id):
```
Comment (gqueiroz) [inline bdc_auth/users/controller.py:113]: Make a more secure code

File: bdc_auth/users/controller.py [hunk 4]
```diff
@@ -110,7 +110,40 @@ def set_role(owner_id, user_id, role_id):
...
+
+    with db.session.begin_nested():
+        for item in user.roles:
+            if item != role:
+                new_roles.append(item)
+        user.roles = new_roles
+
+    db.session.commit()
+
+def get_client_users(user_id=None, client_id=None, page=1, limit=10):
```
Comment (raphaelrpl) [inline bdc_auth/users/controller.py:140]: Do you really need this function?

Check in `bdc_auth.users.controllers.list_users`

File: bdc_auth/roles/views.py [hunk 5]
```diff
@@ -24,6 +24,15 @@ def roles_crud(role_id, user):
     if request.method == 'GET':
         return controller.list_roles()
 
+    if request.method == 'DELETE':
+        try:
+            controller.delete_role(user.id, role_id)
+        except Exception as e:
+            print('erro')
```
Comment (raphaelrpl) [inline bdc_auth/roles/views.py:31]: Please remove `print`

File: bdc_auth/roles/views.py [hunk 6]
```diff
@@ -24,6 +24,15 @@ def roles_crud(role_id, user):
...
+            # TODO treat exceptions and send proper error code.
+            abort(400, '')
```
Comment (raphaelrpl) [inline bdc_auth/roles/views.py:33]: Please remove the try/catch, since the `delete_role` already throws 404 when Role or Client not found.

**Pull request merged**